pyflagser/_utils.py

```diff
@@ -0,0 +1,62 @@
+"""Utility functions for flag matrices"""
+
+import numpy as np
+
+
+def _extract_unweighted_graph(adjacency_matrix):
+    # Extract vertices and give them weight one
+    vertices = np.ones(adjacency_matrix.shape[0], dtype=np.float)
+
+    # Extract edges indices
+    if isinstance(adjacency_matrix, np.ndarray):
+        # Off-diagonal mask
+        mask = np.logical_not(np.eye(adjacency_matrix.shape[0], dtype=bool))
+
+        # Data mask
+        mask = np.logical_and(adjacency_matrix, mask)
+
+        edges = np.argwhere(mask)
+    else:
+        # Data mask
+        mask = np.stack(np.nonzero(adjacency_matrix)).T
+
+        # Removes diagonal elements a posteriori
+        edges = mask[mask[:, 0] != mask[:, 1]]
+
+    # Assign weight one
+    edges = np.hstack([edges, np.ones(edges[:, [0]].shape, dtype=np.int)])
+
+    return vertices, edges
+
+
+def _extract_weighted_graph(adjacency_matrix, max_edge_length):
+    # Extract vertices weights
+    vertices = np.asarray(adjacency_matrix.diagonal())
```

Comment:
I worry for situations in which the adjacency matrix is sparse and rectangular (because all vertices are assumed 0 if not stored and perhaps there are non-infinite edges on one side of the diagonal but not the other). Perhaps we can leave these edge cases for this PR but I wonder if you agree that the issue exists in principle.

Reply:
We should document the fact that the matrix has to be squared and have a warning set up when it's not the case.

I can do that now.

Reply:
We should document the fact that the matrix has to be squared and have a warning set up when it's not the case.

I can do that now.

Reply:
@gtauzin we could do that a, however when we eventually use `flagser_weighted` in `giotto-tda`, we will likely end up duplicating some dimensionality checks (including for square shape, see https://github.com/giotto-ai/giotto-tda/blob/c5bae0438d056ed9977ccc3e9edf34dbe513a6ca/gtda/utils/validation.py#L264 and https://github.com/giotto-ai/giotto-tda/blob/c5bae0438d056ed9977ccc3e9edf34dbe513a6ca/gtda/utils/validation.py#L284).

Reply:
It is really an inexpensive check, I think it is fine.